gnome-initial-setup/gis-page-util.c

```diff
@@ -505,31 +505,34 @@ on_reformatter_exited (GTask  *task,
                        GError *error)
 {
   GisPage *page = GIS_PAGE (g_task_get_source_object (task));
-  GisDriver *driver = page->driver;
 
-  gis_driver_show_window (driver);
+  gis_driver_show_window (page->driver);
 
   if (error == NULL)
     {
       g_task_return_boolean (task, TRUE);
+      return;
     }
-  else
-    {
-      GtkWindow *toplevel = GTK_WINDOW (gtk_widget_get_toplevel (GTK_WIDGET (page)));
-      GtkWidget *message_dialog;
-
-      g_critical ("Error running the reformatter: %s", error->message);
 
-      message_dialog = gtk_message_dialog_new (toplevel,
-                                               GTK_DIALOG_MODAL | GTK_DIALOG_DESTROY_WITH_PARENT,
-                                               GTK_MESSAGE_ERROR,
-                                               GTK_BUTTONS_CLOSE,
-                                               _("Error running the reformatter: %s"), error->message);
-      g_idle_add_full (G_PRIORITY_DEFAULT_IDLE, run_dialog_cb,
-                       message_dialog, (GDestroyNotify) gtk_widget_destroy);
-
-      g_task_return_error (task, g_steal_pointer (&error));
-    }
+  GtkWindow *toplevel = GTK_WINDOW (gtk_widget_get_toplevel (GTK_WIDGET (page)));
+  GtkWidget *message_dialog;
```

Comment:
BTW, reluctantly I usually move the definitions to the top as is common in glib, but if that's not enforced in g-i-s, I am okay with it.

Reply:
I would rather have the `else` block (to introduce a new scope), or use mixed definitions, rather than move the declarations to the top of function.